src/demo.re

```diff
@@ -0,0 +1,34 @@
+let setDict carTypes key origKey data=> {
+  switch (key) {
+    | "none" => {
+      data;
+    }
+    | _ => {
+      let cars = Js.Dict.get carTypes key;
```

Comment:
You can get rid of using Js.Dict completely if you can model it as either a record or Js.t object; the only thing preventing this right now is that you have that electricity field. In general, try to keep things "simple" (in the mathematical definition, aka collection of same things). So use variants to make sure your list of cars contain the same car type in the end. Something like:

```reason
type car = {
  root: string,
  electricity: option int
};
```

Or:

```reason
type eCar = {...};
type gCar = {...};
type car = Electric eCar | Gas gCar
```

etc.

Reply:
Thank you, I plan to leave your comments open so I can use it for reference.